### PR TITLE
EQL: Add explicit 'any where ...' handling

### LINKED ISSUE #51853: EQL: event queries that start with digits fail
`42 where true` throws 
```
org.elasticsearch.xpack.eql.parser.ParsingException: line 1:1: no viable alternative at input '42'
```

### PR DESCRIPTION
**Pull request by rw-access**
Indirectly closes #51853
Relates to #52168 

Added explicit handling for `any where X` so that it converts to the expression `X` instead of `event.type == "any" and X`

Parser handling was already in place:
https://github.com/elastic/elasticsearch/blob/5b4a3a3c07eacbc339cbd4c05a3621d056cc8d60/x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/LogicalPlanBuilder.java#L32-L43

### CODE REVIEW

**Comment by astefan** [comment]
@rw-access it's not clear to me what was the outcome for https://github.com/elastic/elasticsearch/issues/51853. Should `123 where true` be a supported query?

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/parser/LogicalPlanTests.java [hunk 1]
```diff
@@ -21,19 +21,29 @@ public Expression expr(String source) {
         return parser.createExpression(source);
     }
 
+    public Filter filter(Expression expr) {
```
Comment (matriv) [inline x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/parser/LogicalPlanTests.java]: I guess should be private and moved in the beginning or end of class.

**Review by matriv — COMMENTED** [review]
I'm also unfamiliar with the concept of `any WHERE`. Could you please explain?
Should `ANY` be added to the reserverd keywords as well?

**Comment by rw-access** [comment]
@astefan just made the comment on the ticket more direct. It's a "Closed - Won't Fix" type of workaround. Anything that's not an identifier will have to be moved to the right of `where`:
> Instead of accepting `42 where true`, we'll accept `any where event.category == "42"` or `any where event.category == 42`

**Review by matriv — APPROVED** [review]
LGTM

**Review by astefan — APPROVED** [review]
LGTM

**Review by costin — APPROVED** [review]
LGTM

**Comment by rw-access** [comment]
Backport for 7.x a5e82d7fd620ba04f41f989283c6b73d2d4095f1

**Pull request merged**